### ISSUE TITLE
Feedback Request - StrongGrid

### ISSUE THREAD

**Issue by thinkingserious** [sendgrid/sendgrid-csharp#530]
Hello Everyone!

Many of you may have taken a look at the excellent client for SendGrid’s API called [StrongGrid](https://github.com/Jericho/StrongGrid), written by [Jericho](https://github.com/Jericho), an active contributor and supporter of the SendGrid C# community. 

As we started looking ahead at the roadmap of our own API client, we realized that a lot of the functionality we want to add (such as webhook parsing and campaign management just to name a few) is already available in StrongGrid so we engaged in discussions with StrongGrid’s author to see if there’s an opportunity to leverage the work that was done in StrongGrid into SendGrid’s official C# client.

Below are some of the options that were discussed. We want to share these options with the community to give you an opportunity to comment, vote, debate the merits of each option and, in the end, help us shape the future of our official C# client. Please take a moment to provide your thoughts on the following ideas to move forward:

1. Completely replace this library with StrongGrid as a v10 breaking change release.
2. Incorporate as much features from StrongGrid as possible; attempt to limit breaking changes to minimum but some breakage is to be expected.
3. Do not merge these two libraries, continue adding features to the official client organically
4. Some other amazing option we did not think of :)

With Best Regards,

The SendGrid DX Team & @Jericho

**Comment by xt0rted** [comment]
I've never used StrongGrid so based on that I can't really vote for option 1, but I can definitely get behind option 2. It'd also be nice having @Jericho as a maintainer of this project, if they'd want to do it, so new features could get added in faster and to have more eyes for reviewing bug reports & prs.

**Comment by Niladri24dutta** [comment]
@thinkingserious I am ok with option 2 as to merge features into existing sendgrid sdk rather than completely abandon it .Both libraries can exists in parallel . There are many examples in javascript libraries where we can import features from external libraries  like `lodash` or `rxjs` .

**Comment by thinkingserious** [comment]
If you are tagged below, it's because I believe you may have an opinion on this change. If not, my apologizes for the disturbance.

@dylan-asos @0xdeafcafe @owre @JefStat @paritoshmmmec @jonnybee @antimatter96 @nate-fr @warrenkc @ottomatic @sepptruetsch

**Comment by jonnybee** [comment]
I have never used StrongGrid so can't really vote on option 1. It would depend on the legal agreement and continued maintenance for either library. Option 2, to incorporate their features into the existing library is the safest but still may require extra work. You could allow the 2 libraries to co-exist and endorse customer to switch to StrongGrid for more features or new projects.

**Comment by nate-fr** [comment]
I'd vote for 2 - incorporating their feature. I'm against 1 ; a quick glance at their code base doesn't inspire confidence : https://github.com/Jericho/StrongGrid/blob/develop/Source/StrongGrid/WebhookParser.cs

**Comment by owre** [comment]
I probably go for 1 or 2 because I think that StrongGrid is pretty much what sendgrid-csharp should have been from the beginning. Regards

**Comment by spib** [comment]
1 ideally, but 2 is probably more realistic

**Comment by shortstuffsushi** [comment]
2 unless 1 can happen without breaking the API of this lib. Gaining all those featured is great, but since everyone using Send grid proper at the moment doesn't have those features, they aren't going to want go deal with some NuGet update breaking their projects.

**Comment by JefStat** [comment]
I think 2 is the option to go with when comparing the number of impression on each nuget library has ~2.5 million for sendgrid and 24k for stronggrid.

**Comment by paritoshmmmec** [comment]
I would go for this library rather than strong grid. As this library is sponsored by sendgrid itself which provide me more trust using this lib (I am not undermining stronggrid effort). I think we should outline the feature from stronggrid and merge into this lib

**Comment by thinkingserious** [comment]
It looks like option 2 is the preferred choice.

So the next step will be to document the features from StrongGrid that we want to incorporate in this library. Feel free to comment with your favorite missing feature. Thanks!

**Comment by predragbabic** [comment]
Is there any update on this merge?

**Comment by thinkingserious** [comment]
Since there has been no activity on this issue since March 1, 2020, we are closing this issue. Please feel free to reopen or create a new issue if you still require assistance. Thank you!
